### PR TITLE
Mpr121 added options for debounce and touch/release - threshold

### PR DESCRIPTION
**Pull request by mvturnho**
## Description:

```yaml
mpr121:
  id: mpr121_component
  address: 0x5A
  touch_debounce: 5
  release_debounce: 5

binary_sensor:
  - platform: mpr121
    id: touch_key0
    channel: 0
    touch_threshold: 10
    release_threshold: 6
```

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/binary_sensor_map/binary_sensor_map.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#include "binary_sensor_map.h"
```
Comment (OttoWinter) [inline esphome/components/binary_sensor_map/binary_sensor_map.cpp]: The binary_sensor_map got included in this PR - please check your git setup (see contributing site, that explains how to do it best).

File: esphome/components/mpr121/__init__.py [hunk 2]
```diff
@@ -13,10 +14,22 @@
 MULTI_CONF = True
 CONFIG_SCHEMA = cv.Schema({
     cv.GenerateID(): cv.declare_id(MPR121Component),
+    cv.Optional(CONF_RELEASE_DEBOUNCE):  cv.All(cv.Coerce(int), cv.Range(min=0, max=7)),
```
Comment (OttoWinter) [inline esphome/components/mpr121/__init__.py]:
```suggestion
    cv.Optional(CONF_RELEASE_DEBOUNCE): cv.int_range(min=0, max=7),
```

File: esphome/components/mpr121/binary_sensor.py [hunk 3]
```diff
@@ -11,14 +11,24 @@
...
     hub = yield cg.get_variable(config[CONF_MPR121_ID])
+    cg.add(var.set_channel(config[CONF_CHANNEL]))
+    if CONF_TOUCH_THRESHOLD in config:
+        cg.add(var.set_touch_threshold(config[CONF_TOUCH_THRESHOLD]))
+    else:
+        cg.add(var.set_touch_threshold(hub.get_touch_threshold()))
+    if CONF_RELEASE_THRESHOLD in config:
+        cg.add(var.set_release_threshold(config[CONF_RELEASE_THRESHOLD]))
+    else:
+        cg.add(var.set_release_threshold(hub.get_release_threshold()))
```
Comment (OttoWinter) [inline esphome/components/mpr121/binary_sensor.py]: Theoretically the hub could not have set these options yet when this code runs.

I'd say let's remove the per-hub default setting, the user can always use YAML `<<` syntax to copy values in the config.

File: esphome/components/mpr121/__init__.py (hunk 2 above)
Comment (OttoWinter) [inline esphome/components/mpr121/__init__.py]: Define all default values in the schema here. So set this to `default=0` (same for other properties)

File: esphome/const.py [hunk 4]
```diff
@@ -448,6 +448,10 @@
 CONF_WILL_MESSAGE = 'will_message'
 CONF_WINDOW_SIZE = 'window_size'
 CONF_ZERO = 'zero'
+CONF_TOUCH_THRESHOLD = "touch_threshold"
+CONF_RELEASE_THRESHOLD = "release_threshold"
+CONF_TOUCH_DEBOUNCE = "touch_debounce"
+CONF_RELEASE_DEBOUNCE = "release_debounce"
```
Comment (OttoWinter) [inline esphome/const.py]: This list is sorted alphabetically.

But better, these constants will probably only be used in mpr121, so move them into mpr121/__init__.py

File: esphome/components/mpr121/binary_sensor.py (hunk 3 above)
Comment (mvturnho) [reply esphome/components/mpr121/binary_sensor.py]: The code sets the defaults before the channels are created.
When all channels should have alternate threshold the yaml file gets a bit cluttered, that is why I added this code.

**Comment by mvturnho** [comment]
Close this PR for new PR without the binary_sensor_map and changes from this PR. Also updated to latest dev branch

**Comment by mvturnho** [comment]
@OttoWinter what does lint want from me when reporting:

esphome/components/mpr121/__init__.py:6:0: C0412: Imports from package esphome are not grouped (ungrouped-imports)

**Pull request closed without merging**